vignettes/examples.Rmd

```diff
@@ -16,4 +16,146 @@ knitr::opts_chunk$set(
 
 ```{r setup}
 library(tlang)
+library(dplyr)
 ```
+
+# Demography Table
+For this demography table we are going to use `demog_data` an example analysis results table found in the package, which is based on the CDISC pilot data. This dataset has two different row label columns, `rowlbl1` and `rowlbl2` because we are going for a table with group and row labels. There are also two order columns which will be used to set the row order of the output. There is a single column to define our columns (multiple column columns are used when there is column spanning). Finally there is a param column, a value column and a grouping column, `grp`, which we can use for more complex controls.
+
+```{r, echo = FALSE}
+head(demog_data)
+```
+
+The mock we are going to match looks like this:
+
+![Mock deemographics table](images/gt_readme.png)
+
+For this table we have three dose columns, a total column and a p-value column as well as a mix of categorical and continuous analysis. 
+
+The first thing we are going to do when building out the `tfrmt` is specify all our columns 
+```{r}
+tfrmt(
+  # specify columns in the data
+  group = vars(rowlbl1,grp),
+  label = rowlbl2,
+  column = column, 
+  param = param,
+  values = value,
+  sorting_cols = vars(ord1, ord2)) %>% 
+  print_to_gt(demog_data)
+```
+
+While this make a table it isn't a very nice table and defiantly doesn't match the mock. So lets start with formatting all the numbers. To do this we are going to build a `body_plan` to add to our `tfrmt`.  This will be a fairly quick explanation of `body_plan`s but if you like more information see `vignettes("Body Plan")` 
+
+Body plans are made up of a series of `frmt_stucture`s where each `frmt_stucture` represents the formatting of a cell within the table. This will always be applies latest to oldest when applying formatting. This means the first `frmt_stucture` in the `body_plan` should be the most generic. You can use the groups, labels and parameters to specify which formatting applies to which values.  
+
+To start with we are going to use all the rows that are "n (%)" as the default. This way we don't need to list out every row that is an "n (%)" row. These rows are made up of two different values, so we will need to use `frmt_combine`. Next, we can format the continuous variables, which is just a straight forward one value per row so we can just use the label to filter and `frmt` to define the look. Finally, we want format the p-values. This is a bit more complicated because although the group and label will match the other values it needs it own formatting. To so this we will need to specify the parameter in the `frmt_structure` like so `frmt_structure(group_val = ".default", label_val = ".default", p = frmt("x.xx")`. But, we also here we need to make sure it never reports 0 or 1. So we can use `frmt_when` to specify the formatting based on the value. 
```

Comment:
```suggestion
To start, we are going to use all the rows that are "n (%)" as the default. This way we don't need to list out every row that is an "n (%)" row. These rows are made up of two different values, so we will need to use `frmt_combine`. Next, we can format the continuous variables, which is just a straightforward one value per row so we can just use the label to filter and `frmt` to define the look. Finally, we want to format the p-values. This is a bit more complicated, since the p-value sits in the same row as other parameters; therefore the group and label value are not specific enough and we need to be more granular in our specifications. As such, we will need to specify the parameter in the `frmt_structure` like so `frmt_structure(group_val = ".default", label_val = ".default", p = frmt("x.xx")`. Further, we also need to make sure it never displays a rounded p-value of 0 or 1. So we can use `frmt_when` to specify the formatting based on the value. 
```